node/rollback.go

```diff
@@ -20,29 +21,30 @@ const (
 
 func (localNode *LocalNode) maybeRollback(neighbors []*RemoteNode) (bool, error) {
 	currentHeight := chain.DefaultLedger.Store.GetHeight()
-	currentHash, _ := chain.DefaultLedger.Store.GetBlockHash(currentHeight)
+	currentHash, err := chain.DefaultLedger.Store.GetBlockHash(currentHeight)
+	if err != nil {
+		return false, fmt.Errorf("get block hash error: %v", err)
+	}
 
 	majorityBlockHash := localNode.getNeighborsMajorityBlockHashByHeight(currentHeight, neighbors)
 	if majorityBlockHash == common.EmptyUint256 {
-		log.Warningf("Get neighbors majority block hash by height failed")
+		log.Warningf("Get neighbors majority block hash by height failed, skipping rollback.")
 		return false, nil
 	}
 
-	if majorityBlockHash != currentHash {
-		err := fmt.Errorf("Local block hash %s is different from neighbors' majority block hash %s at height %d, rollback needed", currentHash.ToHexString(), majorityBlockHash.ToHexString(), currentHeight)
-		if config.MaxRollbackBlocks == 0 {
-			return false, err
-		}
-		log.Warning(err)
-	} else {
+	if majorityBlockHash == currentHash {
 		return false, nil
 	}
 
+	if config.MaxRollbackBlocks == 0 {
```

Comment:
I guess `config.MaxRollbackBlocks == 0` will always be false.

Reply:
That's true for now, but we have changed MaxRollbackBlocks before, it's just not configurable by users